source/PlayerInfoPanel.cpp

```diff
@@ -190,10 +213,10 @@ bool PlayerInfoPanel::KeyDown(SDL_Keycode key, Uint16 mod, const Command &comman
 		GetUI()->Pop(this);
 	else if(key == 's' || key == SDLK_RETURN || key == SDLK_KP_ENTER)
 	{
-		if(!player.Ships().empty())
+		if(!ships.empty())
 		{
 			GetUI()->Pop(this);
-			GetUI()->Push(new ShipInfoPanel(player, selectedIndex));
+			GetUI()->Push(new ShipInfoPanel(player, selectedIndex, &ships));
```

Comment:
With the requested change to the constructor signature to use `vector<shared_ptr<Ship>> shipList`, I think you should be able to `std::move(ships)` when creating each _____InfoPanel from the other, which should avoid copying.